function-service-impl/src/main/java/com/future/function/service/impl/feature/scoring/QuizServiceImpl.java

```diff
@@ -58,9 +67,17 @@ public Quiz findById(String id) {
       .map(batchService::getBatchByCode)
       .map(
         batch -> quizRepository.findAllByBatchAndDeletedFalse(batch, pageable))
+      .map(this::sortByClosestDeadline)
```

Comment:
I think this is invalid as the sorting is done after getting paged data. A solution would be to use aggregation pipeline in mongodb, an example can be seen at `ActivityBlogRepositoryImpl`

See prev comment